### PR TITLE
Clean up old AI TODOs

### PR DESCRIPTION
**Pull request by mattsc**
This addresses many of the AI TODOs in #3695

### CODE REVIEW

**Comment by CelticMinstrel** [comment]
100000 seems like a pretty low "max" value. Why not use the actual max with `std::numeric_limits<type>::max()`? (Where "type" is substituted by the actual type of the mp_rank variable.)

**Comment by mattsc** [comment]
We're never going to have more than a small number of AIs, so 100000 is plenty from a practical point of view.  But sure, on principle grounds that's better.  Done and squashed to 8a7a5a153f32

File: src/ai/contexts.cpp [hunk 1]
```diff
@@ -1163,7 +1163,6 @@ void readonly_context_impl::recalculate_move_maps() const
 					++it;
 				}
 			}
-		///@todo: shall possible moves be modified as well ?
```
Comment (mattsc) [inline src/ai/contexts.cpp]: Hmm, I just (re)noticed that this (removing moves for passive leaders from `dstsrc` etc) is only done for the leader of the current side. I am still not convinced that it needs to be done for `possible_moves` also, but the explanation in the commit message is not quite right.

Comment (mattsc) [reply src/ai/contexts.cpp]: Looked into this some more, and I think even more so now that `possible_moves` should not have moves of passive leaders removed. It's supposed to be the *possible* moves, after all.  Filtering whether these should be used should happen in the CAs, not here.

If anything, I think that `dstsrc` and `srcdst` should retain those moves also, not the other way around. But there's a lot of weird stuff deep in the AI code here. For example, allied villages are removed from `srcdst` etc. I really don't think it should be done in this way. That's a task for reworking the AI as a whole though.

Bottom line: **I suggest simply removing this TODO comment.**

**Comment by mattsc** [comment]
Just force pushed two more commits: 
ba4202c  `AI code: remove more outdated TODOs`
49e216b  `AI code: remove old version annotations from TODOs`

With that, I am done with this as far as I am concerned.  I'll wait for a day or so for comments, then I'll merge this.
